src/constraints/shake.jl

```diff
@@ -149,13 +190,16 @@ Checks if the velocity constraints are satisfied by the current velocities of `s
 """
 function check_velocity_constraints(sys::System, ca::SHAKE_RATTLE)
```

Comment:
For a GPU system this required `@allowscalar`, maybe there is a way to use a broadcast or reduction here?